packages/client/lib/commands/XADD.ts

```diff
@@ -9,47 +11,47 @@ export interface XAddOptions {
   };
 }
 
-export function pushXAddArguments(
-  args: CommandArguments,
+export function parseXAddArguments(
+  optional: RedisArgument | undefined,
+  parser: CommandParser,
+  key: RedisArgument,
```

Comment:
ponder optional argument up front if better elsewhere?

Reply:
annoying but due to location of NOMKSTREAM (i.e. xadd <key> NOMKSTREAM), I'm not sure of a better way to do it without duplicating code.